src/foam/dao/daoUtils.js

```diff
@@ -113,10 +113,40 @@ foam.CLASS({
   name: 'ArraySink',
   extends: 'foam.dao.AbstractSink',
 
+  constants: {
+    // Dual to outputJSON method.
+    PARSE_JSON: function(json, opt_cls, opt_ctx) {
```

Comment:
How is defining PARSE_JSON for a class different than just implementing a custom initArgs(), which is also responsible for controlling how the object is built from JSON?